src/components/CommentForm.js

```diff
@@ -102,58 +102,67 @@ export default class CommentForm extends React.PureComponent {
     dispatch(updateCommentEditor(storeId, text, newComment))
   }
 
-  delaySetText = debounce(text => this.setText(text), 50)
-
-  render () {
-    const { postId, text, newComment, close, pending } = this.props
-    const { currentUser, isMobile } = this.context
-    const editing = text !== undefined
-    const edit = () => this.setText('')
-    const placeholder = this.props.placeholder || 'Add a comment...'
-
-    const stoppedTyping = () => {
-      if (!newComment) return
-      if (this.socket) this.socket.post(socketUrl(`/noo/post/${postId}/typing`), { isTyping: false })
-    }
+  handleKeyDown = (e) => {
+    const { postId, newComment } = this.props
     const startedTyping = () => {
       if (!newComment) return
       if (this.socket) this.socket.post(socketUrl(`/noo/post/${postId}/typing`), { isTyping: true })
     }
-
-    const stopTyping = debounce(stoppedTyping, STOPPED_TYPING_WAIT_TIME)
     const startTyping = throttle(startedTyping, STARTED_TYPING_INTERVAL, {trailing: false})
-    const handleKeyDown = e => {
-      this.setState({enabled: this.refs.editor.getContent().length > 0})
-      startTyping()
-      onCmdOrCtrlEnter(e => {
-        stoppedTyping()
-        e.preventDefault()
-        this.submit()
-      }, e)
-    }
+    this.setEnabled(this.refs.editor.getContent())
+    startTyping()
+    onCmdOrCtrlEnter(e => {
+      this.stoppedTyping()
+      e.preventDefault()
+      this.submit()
+    }, e)
+  }
 
+  stoppedTyping = () => {
+    const { postId, newComment } = this.props
+    if (!newComment) return
+    if (this.socket) this.socket.post(socketUrl(`/noo/post/${postId}/typing`), { isTyping: false })
+  }
+
+  stopTyping = debounce(this.stoppedTyping, STOPPED_TYPING_WAIT_TIME)
+
+  delaySetText = (text) => {
+    this.setEnabled(text)
+    return debounce(text => this.setText(text), 50)
+  }
+
+  setEnabled = (text) => {
+    this.setState({enabled: text.length > 0})
+  }
+
+  render () {
+    const { text, close, pending } = this.props
+    const { currentUser, isMobile } = this.context
     const { enabled, modifierKey } = this.state
+    const editing = text !== undefined
+    const edit = () => this.setText('')
+    const placeholder = this.props.placeholder || 'Add a comment...'
 
     return <form onSubmit={this.submit} className='comment-form'>
       <Avatar person={currentUser} />
       {editing
         ? <div className='content'>
```

Comment:
Uh yeah-- I am slowly capitulating and this is a weird case (though I don't disagree with it here). have in my sights on getting us on a CircleCI setup that runs a linter pass eventually and so usually bring all files I touch with actual dev work down to 0 Errors...